gef.py

```diff
@@ -736,270 +714,366 @@ class Elf:
-    POWERPC64         = 0x15
-    SPARC             = 0x02
-    SPARC64           = 0x2b
-    AARCH64           = 0xb7
-    RISCV             = 0xf3
-    IA64              = 0x32
-    M68K              = 0x04
-
-    ET_RELOC          = 1
-    ET_EXEC           = 2
-    ET_DYN            = 3
-    ET_CORE           = 4
-
-    OSABI_SYSTEMV     = 0x00
-    OSABI_HPUX        = 0x01
-    OSABI_NETBSD      = 0x02
-    OSABI_LINUX       = 0x03
-    OSABI_SOLARIS     = 0x06
-    OSABI_AIX         = 0x07
-    OSABI_IRIX        = 0x08
-    OSABI_FREEBSD     = 0x09
-    OSABI_OPENBSD     = 0x0C
+    class Abi(enum.Enum):
+        X86_64            = 0x3e
+        X86_32            = 0x03
+        ARM               = 0x28
+        MIPS              = 0x08
+        POWERPC           = 0x14
+        POWERPC64         = 0x15
+        SPARC             = 0x02
+        SPARC64           = 0x2b
+        AARCH64           = 0xb7
+        RISCV             = 0xf3
+        IA64              = 0x32
+        M68K              = 0x04
+
+    class Type(enum.Enum):
+        ET_RELOC          = 1
+        ET_EXEC           = 2
+        ET_DYN            = 3
+        ET_CORE           = 4
+
+    class OsAbi(enum.Enum):
+        SYSTEMV     = 0x00
+        HPUX        = 0x01
+        NETBSD      = 0x02
+        LINUX       = 0x03
+        SOLARIS     = 0x06
+        AIX         = 0x07
+        IRIX        = 0x08
+        FREEBSD     = 0x09
+        OPENBSD     = 0x0C
 
     e_magic: int                = ELF_MAGIC
-    e_class: int                = ELF_32_BITS
-    e_endianness: int           = Endianness.LITTLE_ENDIAN.value
-    e_eiversion: Optional[int]  = None
-    e_osabi: Optional[int]      = None
-    e_abiversion: Optional[int] = None
-    e_pad: Optional[bytes]      = None
-    e_type: int                 = ET_EXEC
-    e_machine: int              = X86_32
-    e_version: Optional[int]    = None
-    e_entry: int                = 0
-    e_phoff : int               = 0
-    e_shoff : int               = 0
-    e_flags: Optional[int]      = None
-    e_ehsize : int              = 0
-    e_phentsize : int           = 0
-    e_phnum : int               = 0
-    e_shentsize : int           = 0
-    e_shnum : int               = 0
-    e_shstrndx : int            = 0
-
-    def __init__(self, elf: str = "", minimalist: bool = False) -> None:
-        """
-        Instantiate an ELF object. The default behavior is to create the object by parsing the ELF file.
+    e_class: Class              = Class.ELF_32_BITS
+    e_endianness: Endianness    = Endianness.LITTLE_ENDIAN
+    e_eiversion: int
+    e_osabi: OsAbi
+    e_abiversion: int
+    e_pad: bytes
+    e_type: Type                = Type.ET_EXEC
+    e_machine: Abi              = Abi.X86_32
+    e_version: int
+    e_entry: int
+    e_phoff : int
+    e_shoff : int
+    e_flags: int
+    e_ehsize : int
+    e_phentsize : int
+    e_phnum : int
+    e_shentsize : int
+    e_shnum : int
+    e_shstrndx : int
```

Comment:
Inconsistent whitespace. It was like this before this PR but I think now's a suitable time to fix it.

```suggestion
    e_phoff: int
    e_shoff: int
    e_flags: int
    e_ehsize: int
    e_phentsize: int
    e_phnum: int
    e_shentsize: int
    e_shnum: int
    e_shstrndx: int
```